include/mesh.tcc

```diff
@@ -1745,3 +1745,24 @@ bool mpm::Mesh<Tdim>::assign_nodal_concentrated_forces(
   }
   return status;
 }
+
+// Initialise the nodal properties' pool
+template <unsigned Tdim>
+bool mpm::Mesh<Tdim>::initialise_nodal_properties() {
+  // Initialise the shared pointer to nodal properties
+  nodal_properties_ = std::make_shared<mpm::NodalProperties>();
```

Comment:
Every time we initialize a nodal property, we recreate this object? Could we split to `create_nodal_properties` and `initialize`, which makes it all zero, so rename to `create_nodal_properties()`